### ISSUE TITLE
How to tune the parameters？

### ISSUE THREAD

**Issue by Jessica0709** [jindongwang/transferlearning#175]
Hello!I am an undergraduate. First of all, thank you for organizing such a perfect transfer learning warehouse.My questions are as follows:

Q1:Why not divide the target domain into training sets and verification sets, as in traditional machine learning, but into test sets as a whole in your code?

Q2:How to tune the parameters of the model?What is the process?For example, for the parameters of mmdloss, Should I choose the cross-validation method to select the parameters or other methods?

**Comment by jindongwang** [comment]
1. Domain adaptation is for adapting the whole target domain. Therefore, the test domain is the target domain. I understand there should be a labeled validation set other than target domain. But in public datasets, people usually don't do that. In real applications, I suggest that you split a validation set. 
2. In order to tune the hyperparams, you can set a unified set of them on all datasets, or you can see the change of the loss for tuning. For other transfer validation methods, you can refer to Long's ICML-19 paper: towards accurate model selection in unsupervised domain adaptation.

**Comment by Jessica0709** [comment]
Thank you very much! And there is a new question.
I run your code on office31_resnet50 dataset which I got it from your warehouse.But I found that in all six of these comparisons the A-distance were 2.0.
In Long's AAAI-18 paper:Multi-Adversarial Domain Adaptation,the resnet50 A-distance is as follows:

![MADA](https://user-images.githubusercontent.com/56783758/83353098-c4242300-a382-11ea-87bc-4a32bdaaa941.jpg)
I'm not sure why they are different.

**Comment by jindongwang** [comment]
The calculation of A-distance depends on how you write it. There is no "accurate" answer. You can find out for yourself. If the code is right, then it's right. But it is only the approximation and not the groundtruth. Different computation of A-distance and different features will definitely lead to different values. Don't be so surprised.

**Comment by Jessica0709** [comment]
I use the code in BDA. All A-distance on this dataset are the same(2.0).Is that normal?
Thank you very much for answering my questions!
![A-distance](https://user-images.githubusercontent.com/56783758/83353402-f6cf1b00-a384-11ea-82d3-b8a3b1d3ba60.png)

**Comment by jindongwang** [comment]
I think this may be a bug since the A-distance in BDA is not used very often. Please try another A-distance code in this code repo: https://github.com/jindongwang/transferlearning/blob/master/code/distance/proxy_a_distance.py
Also, there's a A-distance code in Matlab: https://github.com/jindongwang/transferlearning/blob/master/code/traditional/MEDA/matlab/estimate_mu.m#L24
If the same problem still occurs, you can leave the detailed code information and your screenshot.

**Comment by Jessica0709** [comment]
OK,I will try it. Thank you very much!

**Comment by DF-21D** [comment]
王研究员，您好，非常感谢您的迁移学习资料与代码。我刚刚接触迁移学习，我想问一下关于深度适配网络的问题。训练的准则是什么？是在源域收敛的基础上尽量提高在目标域上的准确率吗？

**Comment by DF-21D** [comment]
训练过程中我把源域和目标域都分成了训练、验证、测试集。训练的时候源域与目标域的训练、验证集参与训练，这样的做法对吗？我还想问一下您提出的BDA可以与CNN结合吗？感谢您在百忙之中回答我的问题。

**Comment by jindongwang** [comment]
BDA当然可以与CNN结合
